tools/integrations/manage.sh

```diff
@@ -217,19 +220,43 @@ cmd_publish() {
   if [[ "airbyte/normalization" == "${image_name}" ]]; then
     echo "Publishing normalization images (version: $versioned_image)"
     GIT_REVISION=$(git rev-parse HEAD)
-    VERSION=$image_version GIT_REVISION=$GIT_REVISION docker-compose -f airbyte-integrations/bases/base-normalization/docker-compose.build.yaml build
-    VERSION=$image_version GIT_REVISION=$GIT_REVISION docker-compose -f airbyte-integrations/bases/base-normalization/docker-compose.build.yaml push
-    VERSION=latest         GIT_REVISION=$GIT_REVISION docker-compose -f airbyte-integrations/bases/base-normalization/docker-compose.build.yaml build
-    VERSION=latest         GIT_REVISION=$GIT_REVISION docker-compose -f airbyte-integrations/bases/base-normalization/docker-compose.build.yaml push
-  else
-    docker tag "$image_name:dev" "$versioned_image"
-    docker tag "$image_name:dev" "$latest_image"
 
-    echo "Publishing new version ($versioned_image)"
-    docker push "$versioned_image"
-    docker push "$latest_image"
+    # Note: "buildx bake" needs to be run within the directory
+    local original_pwd=$PWD
+    cd airbyte-integrations/bases/base-normalization
+
+    VERSION=$image_version GIT_REVISION=$GIT_REVISION docker buildx bake \
+      --set "*.platform=$build_arch"                                     \
+      -f docker-compose.build.yaml                                       \
+      --push
+
+    VERSION=latest GIT_REVISION=$GIT_REVISION docker buildx bake \
+      --set "*.platform=$build_arch"                             \
+      -f docker-compose.build.yaml                               \
+      --push
+
+    cd $original_pwd
+  else
+    # We have to go arch-by-arch locally (see https://github.com/docker/buildx/issues/59 for more info) due to our base images (e.g. airbyte-integrations/bases/base-java)
+    # Alternative local approach @ https://github.com/docker/buildx/issues/301#issuecomment-755164475
+
+    for arch in $(echo $build_arch | sed "s/,/ /g")
+    do
+      echo "building base images for $arch"
+      docker buildx build -t airbyte/integration-base:dev --platform $arch --load airbyte-integrations/bases/base
+      docker buildx build -t airbyte/integration-base-java:dev --platform $arch --load airbyte-integrations/bases/base-java
+
+      local arch_versioned_image=$image_name:`echo $arch | sed "s/\//-/g"`-$image_version
+      echo "Publishing new version ($arch_versioned_image) from $path"
+      docker buildx build -t $arch_versioned_image --platform $arch --push $path
```

Comment:
why don't we directly build the image name? is it because it's not possible to 'add' a second architecture to the same name once the name is 'used'?

Reply:
That's exactly right - because we need to push these tags one-by-one to docker hub, and we don't want the AMD one to overwrite the ARM version, we need this intermediary tags.